Bottom: 97f0bb867f14c12f53c1b0ec28cfd8ca09d18f26
Top:    10ecf8b7802e5761976e9a821167199fbf21756f
Author: Peter Clifton <peter@clifton-electronics.co.uk>
Date:   2014-06-27 00:28:33 +0100

Refresh of patch

---

diff --git a/src/misc.c b/src/misc.c
index 58ed8f5..8d12706 100644
--- a/src/misc.c
+++ b/src/misc.c
@@ -1056,14 +1056,19 @@ ParseGroupString (char *s, LayerGroupType *LayerGroup, int LayerN)
       if (*s == ':')
         s++;
     }
+
+  /* If no explicit solder or component layer group was found in the layer
+   * group string, make group 0 the bottom side, and group 1 the top side.
+   * This is done by assigning the relevant silkscreen layers to those groups.
+   */
   if (!s_set)
-    LayerGroup->Entries[SOLDER_LAYER][LayerGroup->Number[SOLDER_LAYER]++] =
-      LayerN + SOLDER_LAYER;
+    LayerGroup->Entries[0][LayerGroup->Number[0]++] = LayerN + BOTTOM_SILK_LAYER;
   if (!c_set)
-    LayerGroup->
-      Entries[COMPONENT_LAYER][LayerGroup->Number[COMPONENT_LAYER]++] =
-      LayerN + COMPONENT_LAYER;
+    LayerGroup->Entries[1][LayerGroup->Number[1]++] = LayerN + TOP_SILK_LAYER;
 
+  /* Assign a unique layer group to each layer that was not explicitly
+   * assigned a particular group by its presence in the layer group string.
+   */
   for (layer = 0; layer < LayerN && group < LayerN; layer++)
     if (groupnum[layer] == -1)
       {
